ibis/sql/tests/test_compiler.py

```diff
@@ -1296,27 +1297,27 @@ def test_bug_project_multiple_times(self):
         # it works!
         result = to_sql(expr)
         expected = """\
-SELECT t0.`c_name`, t2.`r_name`, t1.`n_name`
-FROM tpch_customer t0
-  INNER JOIN tpch_nation t1
-    ON t0.`c_nationkey` = t1.`n_nationkey`
-  INNER JOIN tpch_region t2
-    ON t1.`n_regionkey` = t2.`r_regionkey`
+WITH t0 AS (
+  SELECT t2.*, t3.`n_name`, t4.`r_name`
+  FROM tpch_customer t2
+    INNER JOIN tpch_nation t3
+      ON t2.`c_nationkey` = t3.`n_nationkey`
+    INNER JOIN tpch_region t4
+      ON t3.`n_regionkey` = t4.`r_regionkey`
+)
+SELECT `c_name`, `r_name`, `n_name`
+FROM t0
```

Comment:
I'm not sure which change causes the CTE extraction, but the queries looks OK.

Reply:
I'm not sure either (it's been some time since I last worked on this particular part of the compiler), but I agree the queries look fine